client/namespace_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+package client_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/proto"
+)
+
+func makeTestSchema(name string) proto.TableSchema {
+	return proto.TableSchema{
+		Table: proto.Table{
+			Name: name,
+		},
+		Columns: []proto.Column{
+			{Name: "id", Type: proto.Column_BYTES},
+			{Name: "name", Type: proto.Column_BYTES},
+			{Name: "title", Type: proto.Column_BYTES},
+		},
+		Indexes: []proto.TableSchema_IndexByName{
+			{Index: proto.Index{Name: "primary", Unique: true},
+				ColumnNames: []string{"id"}},
+		},
+	}
+}
+
+func TestCreateTable(t *testing.T) {
+	s, db := setup()
+	defer s.Stop()
+
+	if _, err := db.DescribeTable("users"); err == nil {
+		t.Fatalf("expected failure, but found success")
+	}
+
+	schema := makeTestSchema("users")
+	if err := db.CreateTable(schema); err != nil {
+		t.Fatal(err)
+	}
+
+	// Table names are case-insensitive.
+	schema2, err := db.DescribeTable("USERS")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(schema, schema2) {
+		t.Errorf("expected %+v, but got %+v", schema, schema2)
+	}
+
+	// Creating the table again should fail as the table already exists.
+	if err := db.CreateTable(schema); err == nil {
```

Comment:
could you use a stronger assertion here on the type of error you expect? this pattern is everywhere in our code but it's very very brittle